pkg/handlers/launch.go

```diff
@@ -214,6 +308,12 @@ func (h *launchHandler) buildEnvVars(payload *launchPayload) (map[string]string,
 }
 
 func (h *launchHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
+	select {
+	case <-h.availableTestRuns:
+	default:
+		http.Error(resp, "Maximum concurrent test runs reached", 429)
```

Comment:
also², what do you think about setting `Retry-After` to a sensible value? hopefully flagger respects that ...

edit: checked and [it uses `retryablehttp`](https://github.com/fluxcd/flagger/blob/adc60596f5a1150d182bd5c3ad66dcd18ff796ea/pkg/controller/webhook.go#L50) which [does respect the header on a 429](https://github.com/hashicorp/go-retryablehttp/blob/1542b31176d3973a6ecbc06c05a2d0df89b59afb/client.go#L544-L566) :+1:

Reply:
hm... good point. I think going with something like 1 minutes should be useful in most scenarios. We could also go with something like the median of all wait-periods so far and even expose that through a histogram or something. Will take a look later :)

Reply:
Done :) I now use the median from all previous execution times and fall back to one minute for the unlikely combination that the limit is reached but we don't have any values yet (e.g. the limit is 1 and we are currently in the first run)